### PR TITLE
feat: New Module avm-res-resource-graph.query

### PR DESCRIPTION
**Pull request by sebassem**
## Description

New Module migrated from CARML

[avm.res.resource-graph.query](https://github.com/sebassem/bicep-registry-modules/actions/workflows/avm.res.resource-graph.query.yml)

### CODE REVIEW

File: avm/res/resource-graph/query/main.bicep [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+resource rgQuery_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: rgQuery
+}
+
```
Comment (AlexanderSehr) [inline avm/res/resource-graph/query/main.bicep:57]: Why is that a dependency? :D

File: avm/res/resource-graph/query/version.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
+    "version": "0.3",
```
Comment (AlexanderSehr) [inline avm/res/resource-graph/query/version.json:3]:
```suggestion
    "version": "0.1",
```

File: avm/res/resource-graph/query/tests/e2e/waf-aligned/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    lock: {
+      kind: 'None'
+    }
+    roleAssignments: [
+      {
+        roleDefinitionIdOrName: 'Reader'
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
+    ]
```
Comment (AlexanderSehr) [inline avm/res/resource-graph/query/tests/e2e/waf-aligned/main.test.bicep:66]:
```suggestion
```

Allows you to remove the `dependencies.bicep` & requires an update of the readme.md

File: avm/res/resource-graph/query/tests/e2e/defaults/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,50 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
+metadata description = '''
+This instance deploys the module with the minimum set of required parameters.
+> **Note:** The test currently implements additional non-required parameters to cater for a test-specific limitation.
+'''
```
Comment (AlexanderSehr) [inline avm/res/resource-graph/query/tests/e2e/defaults/main.test.bicep:7]:
```suggestion
metadata description = 'This instance deploys the module with the minimum set of required parameters.'
```

Actually you don't ;) Requires an update of the readme.md

File: avm/res/resource-graph/query/tests/e2e/defaults/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: location
+    query: 'Resources | limit 10'
+    queryDescription: 'Test query to get the first 10 resources in the subscription.'
```
Comment (AlexanderSehr) [inline avm/res/resource-graph/query/tests/e2e/defaults/main.test.bicep:48]:
```suggestion
```

Should only contain the mandatory parameters.

**Review by AlexanderSehr — COMMENTED** [review]
Cool addition with the `query.kql` file

File: avm/res/resource-graph/query/tests/e2e/waf-aligned/main.test.bicep (hunk 3 above)
Comment (sebassem) [reply avm/res/resource-graph/query/tests/e2e/waf-aligned/main.test.bicep:66]: @AlexanderSehr  should I remove this from the max test as well ?

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**